cloud/amazon/emr.py

```diff
@@ -0,0 +1,522 @@
+        """
+        :returns map: a map of key/value pairs
+        """
+
+        tag_dict = {}
+
+        for tag in self.tags:
+            name = tag['name']
+            value = tag['value']
+            tag_dict[name] = value
+
+        return tag_dict
+
+    def run_jobflow(self):
+        """
+        :returns EMR object: The ID of the created job flow
+        """
+
+        if self.vpc_subnet_id:
+            my_api_params = {'Instances.Ec2SubnetId': self.vpc_subnet_id}
+        else:
+            my_api_params = None
+
+        self.jobflow_id = self.conn.run_jobflow(name=self.cluster_name,
+                                                ec2_keyname=self.ec2_keyname,
+                                                keep_alive=self.keep_alive,
+                                                enable_debugging=self.enable_debugging,
+                                                log_uri='{0}/{1}'.format(self.log_uri, datetime.now().strftime("%Y-%m-%d %H-%M-%S")),
+                                                api_params=my_api_params,
+                                                ami_version=self.ami_version,
+                                                job_flow_role=self.jobflow_role,
+                                                service_role=self.service_role,
+                                                visible_to_all_users=self.visible_to_all_users,
+                                                instance_groups=self.jobflow_instance_groups,
+                                                bootstrap_actions=self.jobflow_bootstrap_actions,
+                                                steps=self.jobflow_jar_steps)
+
+        if self.cluster_tags:
+            self.conn.add_tags(self.jobflow_id, self.cluster_tags)
+        # add termination protection if necessary
+        if self.termination_protection:
+            self.conn.set_termination_protection(self.jobflow_id, self.termination_protection)
+
+    def add_jobflow_steps(self):
+        """
+        Adds jobs provided in the playbook to an existing cluster/jobflow.
+
+        :returns: nada
+        """
+
+        self.conn.add_jobflow_steps(self.jobflow_id, self.jobflow_jar_steps)
+
+    def termiante_jobflow(self):
+        """
+        Terminates the jobflow.
+        """
+        if self.force_termination:
+            self.conn.set_termination_protection(self.jobflow_id, False)
+        self.conn.terminate_jobflow(self.jobflow_id)
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            ec2_keyname={'default': None},
+            cluster_name={},
+            jobflow_id={'default': None},
+            log_uri={},
+            ec2_secret_key={'default': None, 'aliases': ['aws_secret_key', 'secret_key'], 'no_log': True},
+            ec2_access_key={'default': None, 'aliases': ['aws_access_key', 'access_key']},
+            region={'default': None, 'aliases': ['aws_region', 'ec2_region'], 'choices': AWS_REGIONS},
```

Comment:
Where is `AWS_REGIONS` defined ? I have an error about this when trying to use the module
```
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: NameError: global name 'AWS_REGIONS' is not defined
fatal: [localhost]: FAILED! => {"changed": false, "failed": true, "module_stderr": "Traceback (most recent call last):
  File "/var/folders/ll/by_nmkg14ng_kvyn_wty1dlc0000gn/T/ansible_qedzmd/ansible_module_emr.py", line 526, in <module>
    main()
  File "/var/folders/ll/by_nmkg14ng_kvyn_wty1dlc0000gn/T/ansible_qedzmd/ansible_module_emr.py", line 483, in main
    region={'default': None, 'aliases': ['aws_region', 'ec2_region'], 'choices': AWS_REGIONS},
NameError: global name 'AWS_REGIONS' is not defined
", "module_stdout": "", "msg": "MODULE FAILURE", "parsed": false}
```